app/helpers/application_helper.rb

```diff
@@ -260,6 +244,18 @@ def view_to_url(view, parent = nil)
     end
   end
 
+  def class_to_db(cname, parent)
+    return unless cname
+    case
```

Comment:
I agree, that's why I like @abonas' suggestion.  It's more of an extract method.